idaes/core/base/control_volume0d.py

```diff
@@ -341,13 +341,31 @@ def _add_material_balance_common(
 
         # Material holdup and accumulation
         if has_holdup:
+            if (
+                self.properties_in[
+                    self.flowsheet().time.first()
+                ].get_material_flow_basis()
+                == MaterialFlowBasis.mass
+            ):
+                holdup_units = units("mass")
+            elif (
+                self.properties_in[
+                    self.flowsheet().time.first()
+                ].get_material_flow_basis()
+                == MaterialFlowBasis.molar
+            ):
+                holdup_units = units("amount")
+            else:
+                holdup_units = None
+                # volume units will need to be updated to None for unit consistency in material_holdup_calculation
+                self.volume._units = None
```

Comment:
```suggestion
                self.volume._units = None
```
@andrewlee94 - I am guessing you'd opt to remove this bit.

Reply:
Yes - I'd leave that as is and accept that units will not be consistent. We know what units volume will have from the property package, so we can assign them here.

For example, a flow-on effect of this would be that units would be inconsistent for rate based reactions (where we expect `extent = volume*rate`). I.e. this is just pushing the error around to a different location. Also note that with a flow basis of `other` we cannot convert a united reaction rate to the correct units either, so there is a potential second unit inconsistency that arises with flow basis of `other` (hence my comment that when a user selects `other` they are taking responsibility for unit consistency themselves).